api/v1/harborcluster_types.go

```diff
@@ -16,25 +16,321 @@ limitations under the License.
+	Version string `json:"version"`
+
+	// The url exposed to clients to access harbor
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^https?://.*$"
+	PublicURL string `json:"publicURL"`
+
+	// Password for the root admin
+	// +kubebuilder:validation:Required
+	AdminPasswordSecret string `json:"adminPasswordSecret"`
+
+	// Secret reference for the TLS certs
+	// +optional
+	TLSSecre string `json:"tlsSecret"`
+
+	// The issuer for Harbor certificates.
+	// If the 'kind' field is not set, or set to 'Issuer', an Issuer resource
+	// with the given name in the same namespace as the Certificate will be used.
+	// If the 'kind' field is set to 'ClusterIssuer', a ClusterIssuer with the
+	// provided name will be used.
+	// The 'name' field in this stanza is required at all times.
+	CertificateIssuerRef cmmeta.ObjectReference `json:"certificateIssuerRef"`
+
+	// Indicates that the harbor is paused.
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+
+	// The Maximum priority. Deployments may be created with priority in interval ] priority - 100 ; priority ]
+	// +kubebuilder:validation:Optional
+	Priority *int32 `json:"priority,omitempty"`
+
+	// Pod instance number
+	// +kubebuilder:validation:Required
+	Replicas int `json:"replicas,omitempty"`
+
+	// Source registry of images
+	ImageSource *ImageSource `json:"imageSource,omitempty"`
+
+	// Extra configuration options for jobservices
+	// +optional
+	JobService *JobService `json:"jobService,omitempty"`
+
+	// Extra configuration options for clair scanner
+	// +optional
+	Clair *Clair `json:"clair,omitempty"`
+
+	// Extra configuration options for trivy scanner
+	Trivy *Trivy `json:"trivy,omitempty"`
+
+	// Extra configuration options for chartmeseum
+	ChartMuseum *ChartMuseum `json:"chartMuseum,omitempty"`
+
+	// Cache service(Redis) configurations might be external redis services or inCluster redis services
+	// +kubebuilder:validation:Required
+	Redis *Redis `json:"redis,omitempty"`
+
+	// database service (PostgresSQL) configuration
+	// +kubebuilder:validation:Required
+	Database *Database `json:"Database,omitempty"`
+
+	// Storage service configurations. Might be external cloud storage services or inCluster storage (minIO)
+	// +kubebuilder:validation:Required
+	Stroage *Storage `json:"storage,omitempty"`
+}
+
+type Storage struct {
+	// set the kind of which storage service to be used. Set the kind as "azure", "gcs", "s3", "oss", "swift" or "inCluster", and fill the information.
+	// in the options section. inCluster indicates the local storage service of harbor-cluster. We use minIO as a default built-in object storage service.
+	Kind string `json:"kind,omitempty"`
+
+	// inCLuster options.
+	InCluster *InClusterOptions `json:"options,omitempty"`
```

Comment:
json should not be options, should be inCluster

Reply:
And it's better use struct InCluster instead of InClusterOptions, that would be more concordant